models/base.js

```diff
@@ -0,0 +1,67 @@
+const Promise = require("bluebird");
+const db = require("../server/db");
+
+class BaseModel {
+  constructor(info) {
+    
+    this._info = info
+
+    info.columns.map(name => {
+      this[name] = info.data[name]
+    })
+  }
+
+  static _checkModelValidity() {
+    this._checkForTableName()
+    this._checkForPrimaryId()
+  }
+
+  static _checkForTableName() {
+      if(!this.tableName) {
+        throw new Error("This model needs a tableName!")
+      }    
+  }
+
+  static _checkForPrimaryId() {
+      if(!this.primaryId) {
+        throw new Error("This model needs a primaryId!")
+      }    
+  }
+
+  static _getSingleResponse(objects, fulfill, reject) {
+    if (!objects || objects.length === 0) {
+      reject(new Error("nothing returned from db"))
+    } else {
+      // return class instance
+      let instance = new this(objects[0])
```

Comment:
`info` is undocumented and very important. Every new model has to pass a correctly formatted `info` object to the BaseModel constructor. Would argue this isn't very Rails-esque and is kind of a hack to get all the values we need for the model. Will discuss on the call as I would be open to an alternate way to handle this.